core/src/main/java/org/apache/iceberg/util/ParallelIterable.java

```diff
@@ -67,10 +70,17 @@ private ParallelIterator(
                             try (Closeable ignored =
                                 (iterable instanceof Closeable) ? (Closeable) iterable : () -> {}) {
                               for (T item : iterable) {
-                                queue.add(item);
+                                if (closed) {
+                                  queue.clear();
+                                  return;
+                                }
+                                queue.put(item);
                               }
                             } catch (IOException e) {
                               throw new RuntimeIOException(e, "Failed to close iterable");
+                            } catch (InterruptedException e) {
+                              throw new RuntimeException(
```

Comment:
Do we actually want to throw here or just log?  It seems like you'll have to deal with an unnecessary/non-actionable exception when you're really just trying to cancel the iteration.  Maybe just turn this into an info log message?

Reply:
@danielcweeks If we had closed this iterator, we don't care of throw error or log. I don't use log only because this class don't have log, I just follow it. Maybe we can add log to this class.